mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -795,8 +799,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       require(a.length == k)
       copyToDouble(a)
       blas.dspr(upper, k, c, da, 1, ata)
-      if (b != 0.0) {
-        blas.daxpy(k, c * b, da, 1, atb, 1)
+      if (Math.abs(b) > Double.MinPositiveValue) {
```

Comment:
How does this differ from `!= 0.0`? I get that it differs for `Double.MinPositiveValue` but why is that important?

You're right that the condition was `b > 0` before on purpose, though I think this is trying to handle explicit/implicit cases